src/hints/execution.rs

```diff
@@ -1733,6 +1734,47 @@ pub fn get_old_block_number_and_hash(
     Ok(())
 }
 
+pub const FETCH_RESULT: &str = indoc! {r#"
+    # Fetch the result, up to 100 elements.
+    result = memory.get_range(ids.retdata, min(100, ids.retdata_size))
+
+    if result != [ids.VALIDATED]:
+        print("Invalid return value from __validate__:")
+        print(f"  Size: {ids.retdata_size}")
+        print(f"  Result (at most 100 elements): {result}")"#
+};
+
+pub fn fetch_result(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    // Fetch the result, up to 100 elements.
+    let retdata = get_ptr_from_var_name(vars::ids::RETDATA, vm, ids_data, ap_tracking)?;
+    let retdata_size = get_integer_from_var_name(vars::ids::RETDATA_SIZE, vm, ids_data, ap_tracking)?;
+
+    // validated is the string "VALID" translated to a felt.
+    let validated = constants
```

Comment:
Just realized that there is a `get_constant_from_var_name` in the vm.

Reply:
Missed it as well! I'll replace it.

Reply:
Oh actually this does not behave as expected. It takes the name of constant (ex: `VALIDATED`) and looks for it through the whole hashmap to find an entry that matches by splitting the path. Very expensive operation that we can entirely avoid. I will add a helper function instead.